File: Upcoder/Lập trình cơ bản/5.Mảng/Đĩa Nhạc/DISC.py
def main():
    n, m = map(int, input().split())
    disks = list(range(1, n + 1))  # Khởi tạo mảng đĩa nhạc từ 1 đến n

    actions = list(map(int, input().split()))  # Nhập chuỗi các số đĩa cần rút
    
    for x in actions:
        # Tìm và xóa đĩa x
        disks.remove(x)
        # Đưa đĩa x lên đầu
        disks.insert(0, x)

    # In ra thứ tự các đĩa nhạc
    print(" ".join(map(str, disks)))

if __name__ == "__main__":
    main()

File: Upcoder/Lập trình cơ bản/5.Mảng/Đĩa Nhạc/DISC.cs
using System;
using System.Collections.Generic;
namespace Dianhac
{
    public class Program {
        static void Main() {
            var inputs = Console.ReadLine().Split();
            int n = int.Parse(inputs[0]); // số lượng đĩa

            List<int> disks = new List<int>();
            for (int i = 1; i <= n; i++) {
                disks.Add(i);
            }

            // Nhập chuỗi các số đĩa cần rút
            var actions = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);

            foreach (var x in actions) {
                // Tìm và xóa đĩa x
                disks.Remove(x);
                // Đưa đĩa x lên trên cùng
                disks.Insert(0, x);
            }

            // In ra thứ tự các đĩa nhạc
            Console.WriteLine(string.Join(" ", disks));
        }
    }
}
